src/components/CreateMenu.js

```diff
@@ -25,53 +26,85 @@ const propTypes = {
     ...windowDimensionsPropTypes,
 };
 
-const CreateMenu = (props) => {
-    // This format allows to set individual callbacks to each item
-    // while including mutual callbacks first
-    const menuItemData = [
-        {icon: ChatBubble, text: 'New Chat', onPress: () => {}},
-        {icon: Users, text: 'New Group', onPress: () => {}},
-    ].map(item => ({
-        ...item,
-        onPress: () => {
-            props.onItemSelected();
-            item.onPress();
-        },
-    }));
+class CreateMenu extends React.Component {
+    constructor(props) {
+        super(props);
 
-    return (
-        <Modal
-            onClose={props.onClose}
-            isVisible={props.isVisible}
-            type={
-                props.isSmallScreenWidth
-                    ? CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED
-                    : CONST.MODAL.MODAL_TYPE.POPOVER
-            }
-        >
-            {menuItemData.map(({icon, text, onPress}) => (
-                <Pressable
-                    key={text}
-                    onPress={onPress}
-                    style={({hovered}) => ([
-                        styles.createMenuItem,
-                        {backgroundColor: hovered ? themeColors.buttonHoveredBG : colors.transparent},
-                    ])}
-                >
-                    <View style={styles.createMenuIcon}>
-                        <Icon src={icon} />
-                    </View>
-                    <View style={styles.justifyContentCenter}>
-                        <Text style={[styles.createMenuText, styles.ml3]}>
-                            {text}
-                        </Text>
-                    </View>
-                </Pressable>
-            ))}
-        </Modal>
-    );
-};
+        this.state = {
+            onModalHide: () => {},
+        };
+    }
+
+    /**
+     * Sets a new function to execute when the modal hides
+     */
+    setOnModalHide() {
+        this.setState({onModalHide: () => redirect(ROUTES.NEW_GROUP)});
```

Comment:
It makes a lot of sense to not use state here, just a reference. We definitely don't need a re-render.
Just to make sure, is it what you meant?
```javascript
(...)
this.setOnModalHide = this.setOnModalHide.bind(this);
this.resetOnModalHide = this.resetOnModalHide.bind(this);
this.onModalHide = () => {};
(...)
setOnModalHide() {
    this.onModalHide = () => redirect(ROUTES.NEW_GROUP);
}
(...)
```

Reply:
The above, plus adjusting everything related to the changes.

Reply:
I was thinking of the `setOnModalHide` to take a function as a param as well.

Reply:
>take a function as a param as well, so it can be reused if necessary

Will leave this up to you. It would also be fine to wait until something wants to reuse it.

Reply:
Left it as it is.